src/aiq/agent/react_agent/register.py

```diff
@@ -101,15 +107,15 @@ async def _response_fn(input_message: AIQChatRequest) -> AIQChatResponse:
             state = ReActGraphState(messages=messages)
 
             # run the ReAct Agent Graph
-            state = await graph.ainvoke(state, config={'recursion_limit': (config.max_iterations + 1) * 2})
+            state = await graph.ainvoke(state, config={'recursion_limit': (config.max_tool_calls + 1) * 2})
             # setting recursion_limit: 4 allows 1 tool call
             #   - allows the ReAct Agent to perform 1 cycle / call 1 single tool,
             #   - but stops the agent when it tries to call a tool a second time
 
             # get and return the output from the state
             state = ReActGraphState(**state)
             output_message = state.messages[-1]  # pylint: disable=E1136
-            return AIQChatResponse.from_string(output_message.content)
+            return AIQChatResponse.from_string(str(output_message.content))
```

Comment:
I'm worried about unintended consequences of this. What is the content of `output_message.content`?